gnc/sub8_perception/nodes/ocr_server.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python
+import cv2
+import cv2.cv as cv
+import numpy as np
+import sys
+import rospy
+import image_geometry
+import sub8_ros_tools
+from sub8_msgs.srv import OcrRequest
+from std_msgs.msg import Header, String
+from geometry_msgs.msg import Pose2D
+
+import tesseract
+from matplotlib import pyplot as plt
+
+class Ocr:
+
+    def __init__(self):
+        rospy.sleep(1.0)
+        self.last_image = None
+        self.last_draw_image = None
+        self.last_image_time = None
+        self.camera_model = None
+        self.white_list = None
+        self.ocr_service = rospy.Service('vision/ocr', OcrRequest, self.request_characters)
+        self.image_sub = sub8_ros_tools.Image_Subscriber('/forward_camera/image_rect_color', self.image_cb)
+
+    def request_characters(self, srv):
+        self.white_list = srv.target_name
+        if (self.last_image != None):
+            print 'requesting', srv
+            response = self.ocr()
+
+            if response is False:
```

Comment:
Right, but actually that whole if else block is redundant. The None condition is fine. Ill fix that now.